contracts/base/amm/services/AmmCloseSwapServiceBaseV1.sol

```diff
@@ -486,11 +509,41 @@ abstract contract AmmCloseSwapServiceBaseV1 is IAmmCloseSwapService {
                 unwindingFeeTreasuryPortionRate: unwindingFeeTreasuryPortionRate,
                 maxLengthOfLiquidatedSwapsPerLeg: liquidationLegLimit,
                 timeBeforeMaturityAllowedToCloseSwapByCommunity: timeBeforeMaturityAllowedToCloseSwapByCommunity,
-                timeBeforeMaturityAllowedToCloseSwapByBuyer: timeBeforeMaturityAllowedToCloseSwapByBuyer,
+                timeBeforeMaturityAllowedToCloseSwapByBuyerTenor28days: timeBeforeMaturityAllowedToCloseSwapByBuyerTenor28days,
+                timeBeforeMaturityAllowedToCloseSwapByBuyerTenor60days: timeBeforeMaturityAllowedToCloseSwapByBuyerTenor60days,
+                timeBeforeMaturityAllowedToCloseSwapByBuyerTenor90days: timeBeforeMaturityAllowedToCloseSwapByBuyerTenor90days,
                 minLiquidationThresholdToCloseBeforeMaturityByCommunity: minLiquidationThresholdToCloseBeforeMaturityByCommunity,
                 minLiquidationThresholdToCloseBeforeMaturityByBuyer: minLiquidationThresholdToCloseBeforeMaturityByBuyer,
                 minLeverage: minLeverage,
-                timeAfterOpenAllowedToCloseSwapWithUnwinding: timeAfterOpenAllowedToCloseSwapWithUnwinding
+                timeAfterOpenAllowedToCloseSwapWithUnwindingTenor28days: timeAfterOpenAllowedToCloseSwapWithUnwindingTenor28days,
+                timeAfterOpenAllowedToCloseSwapWithUnwindingTenor60days: timeAfterOpenAllowedToCloseSwapWithUnwindingTenor60days,
+                timeAfterOpenAllowedToCloseSwapWithUnwindingTenor90days: timeAfterOpenAllowedToCloseSwapWithUnwindingTenor90days
             });
     }
+
+    function _getTimeAfterOpenAllowedToCloseSwapWithUnwinding(
+        IporTypes.SwapTenor tenor
+    ) internal view returns (uint256) {
+        if (tenor == IporTypes.SwapTenor.DAYS_28) {
+            return timeAfterOpenAllowedToCloseSwapWithUnwindingTenor28days;
+        } else if (tenor == IporTypes.SwapTenor.DAYS_60) {
+            return timeAfterOpenAllowedToCloseSwapWithUnwindingTenor60days;
+        } else if (tenor == IporTypes.SwapTenor.DAYS_90) {
+            return timeAfterOpenAllowedToCloseSwapWithUnwindingTenor90days;
+        } else {
+            revert(AmmErrors.UNSUPPORTED_SWAP_TENOR);
+        }
+    }
+
+    function _getTimeBeforeMaturityAllowedToCloseSwapByBuyer(IporTypes.SwapTenor tenor) internal view returns (uint256) {
+        if (tenor == IporTypes.SwapTenor.DAYS_28) {
```

Comment:
this if-else logic is duplicated in lib

Reply:
not exactly because here timeAfterOpenAllowedToCloseSwapWithUnwindingTenorXXX is from that contract but in lib is taken from structure AmmCloseSwapServicePoolConfiguration